app/views/layouts/application.html.erb

```diff
@@ -17,13 +17,24 @@
       <%= link_to('New', posts_url) %> |
       <%= link_to('Submit Post', new_post_url) %>
       <span class="accounts">
-        <% if user_signed_in? %>
-          <%= link_to(current_user.email, user_posts_path(current_user)) %> |
-          <%= link_to('Logout', destroy_user_session_path, method: :delete) %>
-        <% else %>
-          <%= link_to('Log In', new_user_session_path) %> |
-          <%= link_to('Create Account', new_user_registration_path) %>
-        <% end %>
+        <p>
+          <% if user_signed_in? %>
+            <%= link_to(current_user.email, user_posts_path(current_user)) %>
+            <% karma = 0 %>
+            <% current_user.posts.each do |p| %>
+              <% karma += p.votes.sum('vote_type') %>
+            <% end %>
+            <% current_user.comments.each do |c| %>
+              <% karma += c.votes.sum('vote_type') %>
+            <% end %>
+            (<%= karma %>)
+            |
+            <%= link_to('Logout', destroy_user_session_path, method: :delete) %>
```

Comment:
This kind of code definitely shouldn't be in the view. Calculations are application logic, and views should really only have display logic. This would probably be better in a model. Think about how you would want to call it. Maybe `current_user.karma`, then write the method that does that.